ops/model.py

```diff
@@ -42,6 +43,22 @@ def get_relation(self, relation_name, relation_id=None):
         """
         return self.relations._get_unique(relation_name, relation_id)
 
+    def get_binding(self, binding_key):
```

Comment:
"Binding names" are names of relations or extra-bindings and the `NetworkInfo` lookup is influenced by the calling unit (implicit in network-get calls), binding name and relation ID (if present).

https://github.com/juju/juju/blob/juju-2.7.1/worker/uniter/runner/jujuc/network-get.go#L116
https://github.com/juju/juju/blob/juju-2.7.1/network/network.go#L242-L253
https://github.com/juju/juju/blob/juju-2.7.1/apiserver/params/network.go#L859-L869

Identifying networking information is not done merely by a relation name but also by a relation ID. This might seem counter-intuitive, however, relation names/endpoints and their binding to spaces exist before relation instances themselves exist (calling `network-get <binding-name>` is possible without any relation-id) and having a relation-id affects what is placed into networking info.

https://github.com/juju/juju/blob/juju-2.7.1/apiserver/facades/agent/uniter/networkinfo.go#L111-L145
https://github.com/juju/juju/blob/juju-2.7.1/apiserver/facades/agent/uniter/networkinfo.go#L244-L275
https://github.com/juju/juju/blob/juju-2.7.1/apiserver/facades/agent/uniter/networkinfo.go#L313-L334

That was the motivation behind `binding_key`, however, looking at Juju now I don't see separate bindings objects for each relation ID - only conditional logic for populating network info.

https://github.com/juju/juju/blob/juju-2.7.1/state/endpoint_bindings.go#L483-L499

So it might be that we need to return `NetworkInfo` objects as opposed to `Binding` objects.